### PR TITLE
Mitra/Add reusable button component

### PR DESCRIPTION
**Pull request by mitra-deriv**
Changes:

Co-authored-by: @farzin-deriv 

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by habib-deriv — APPROVED** [review]
LGTM

File: src/components/custom/_button.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    label: TString
+    onClick?: VoidFunction
+    primary?: boolean
+    secondary?: boolean
+    tertiary?: boolean
+    outline?: boolean
+    hero?: boolean
+}
+
+const Label = styled.span<Omit<TProps, 'label' | 'onClick'>>`
```
Comment (farzin-deriv) [inline src/components/custom/_button.tsx:15]:
```suggestion

type TLabelProps = Omit<TProps, 'label' | 'onClick'>

type TContainerProps = Omit<TProps, 'label'>

const Label = styled.span<TLabelProps>`...`

const Container = styled.div<TContainerProps>`...`
```
@mitra-deriv Just a suggestion for better readability 🙇🏻

File: src/components/custom/_button.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            primary && outline ? 'background-color: var(--color-red)' : ''};
+        ${({ primary, outline }) =>
+            primary && !outline ? 'background-color: var(--color-red-6)' : ''};
+    }
+    &:hover ${Label} {
+        ${({ primary, outline }) => (primary && outline ? 'color: var(--color-white)' : '')};
+    }
+`
+
+const Button: React.FC<React.PropsWithChildren<TProps>> = ({
```
Comment (farzin-deriv) [inline src/components/custom/_button.tsx:63]:
```suggestion
const Button: React.FC<TProps> = ({
```
@mitra-deriv The component doesn’t accept any children.

File: src/components/custom/_button.tsx (hunk 1 above)
Comment (mitra-deriv) [reply src/components/custom/_button.tsx:15]: @farzin Thank you for the suggestions I have a Todo which I need to address them later will apply your comments in that branch, Once done in that branch will resolve these comments. 🌹 🙏

Comment (mitra-deriv) [reply src/components/custom/_button.tsx:15]: Solved in this PR #4114

File: src/components/custom/_button.tsx (hunk 2 above)
Comment (mitra-deriv) [reply src/components/custom/_button.tsx:63]: Solved in this PR #4114

**Pull request merged**